### PR TITLE
Add an option to aggressively delete things the tarball doesn't need and turn it on for CI.

### PR DESCRIPTION
**Pull request by crummel**
This should not be used normally, as it will break the repo and delete local work for the sake of minimizing space.

### CODE REVIEW

**Comment by crummel** [comment]
This take gives us 3GB to spare at the point we're using the most space, so still not great, but at least as an interim step it gets builds off the ground.  OSX is a separate issue I'm still working on.

File: build-source-tarball.sh [hunk 1]
```diff
@@ -124,6 +176,11 @@ for srcDir in `find bin/src -name '.git' -type d`; do
 done
 popd
 
+# now we don't need bin/src at all
+if [ $MINIMIZE_DISK_USAGE -eq 1 ]; then
+    rm -rf "$SCRIPT_ROOT/bin/src/"
```
Comment (dseefeld) [inline build-source-tarball.sh]: I think this deletes all production repo binlogs before they get copied in the next step.

Comment (crummel) [reply build-source-tarball.sh]: Nice catch, thanks.  Removed this part.

**Comment by crummel** [comment]
OSX failure is expected; see https://github.com/dotnet/source-build/pull/1262 for that fix.

**Pull request merged**